### PR TITLE
Navigation a11y edits

### PR DESCRIPTION
**Pull request by adriencyberspace**
Closes https://www.notion.so/exygy/Erroneous-tab-navigating-submenu-while-hidden-7d14c7c519f14cd4898e3a4047bc7b35 and https://www.notion.so/exygy/Dropdown-menu-should-close-after-clicking-away-60adab4a674d4b9ea709a794cee3bfec

This PR:
- Refactors Navigation components into separate files in a new `ui/Navigation` directory
- Adds fixes to desktop nav that makes it so when a user clicks or tabs away or clicks a link, the submenu closes
- Adds fixes to the mobile nav that makes it so when a user clicks a link, the whole sidebar menu closes
- Fixes the bug where user tab navigated through the hidden mobile nav items on desktop

### CODE REVIEW

File: app/components/ui/Navigation/MobileMenuItems.tsx [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+  ExtractedNavigationMenusFromNavigationResponse,
+} from "../../../models/Strapi";
+
+function menuItemHasLinks(
+  menuItem: ExtractedNavigationMenusFromNavigationResponse[number]
+): menuItem is StrapiModel.NavigationMenu {
+  return "link" in menuItem;
+}
+
+const MobileMenuItems = ({
```
Comment (adriencyberspace) [inline app/components/ui/Navigation/MobileMenuItems.tsx:16]: Pulled this out of `Navigation`

File: app/components/ui/Navigation/Navigation.tsx [hunk 2]
```diff
@@ -0,0 +1,96 @@
+import React, { useState } from "react";
```
Comment (adriencyberspace) [inline app/components/ui/Navigation/Navigation.tsx:1]: Pulled all components into separate files

File: app/components/ui/Navigation/MobileNavigation.module.scss [hunk 3]
```diff
@@ -1,8 +1,14 @@
-@import "../../styles/utils/_helpers.scss";
-@import "../../styles/utils/_colors.scss";
+@import "../../../styles/utils/_helpers.scss";
+@import "../../../styles/utils/_colors.scss";
 
 // Edit naming conventions for clarity
 
+.sidebar {
```
Comment (adriencyberspace) [inline app/components/ui/Navigation/MobileNavigation.module.scss:6]: Hide sidebar items from tab navigation while on desktop

File: app/components/search/Sidebar/Sidebar.module.scss [hunk 4]
```diff
@@ -35,7 +35,6 @@
     height: auto;
     width: 60%;
     padding: 0;
-    z-index: $z-index-filters-menu;
```
Comment (rosschapman) [inline app/components/search/Sidebar/Sidebar.module.scss:38]: Yay. Not relying on z-index is usually good.

**Review by rosschapman — APPROVED** [review]
🔥 🚀

**Pull request merged**